lib/galaxy/auth/__init__.py

```diff
@@ -66,7 +66,8 @@ def check_auto_registration(self, trans, login, password, no_password_check=Fals
                     try:
                         auth_return = parse_auth_results(trans, auth_results, options)
                     except Conflict:
-                        break
+                        log.exception(Conflict)
+                        raise
```

Comment:
What about:
```
log.exception(Conflict)
break
```

Reply:
Why would you hide the exception to the uppers layers , instead of allowing them to handle it, for instance with a custom error page ? In this specific case, a red flash banner with an authentication failure message is gently display to the user.

Reply:
Without the `raise` I think you should get a "No such user or invalid password." message, what do you get with the `raise` instead?

Reply:
A blue banner with the  'Conflict' message (self.err_msg) is display , most likely the : "Cannot make unique username" raise in 'parse_auth_results'.  A 'break' indeed give a red banner with a "No such user or invalid password" message.  The 409 error state of the 'Conflict' exception does'nt seems to matter in both cases.